### PR TITLE
Issue/PR templates adjustments, CoC

### PR DESCRIPTION
**Pull request by chrisarcand**
## Description

Adds the usual HashiCorp Community Guidelines CoC + an overhaul of issue templates. The issue templates are [similar to Terraform CLI's](https://github.com/hashicorp/terraform/tree/main/.github/ISSUE_TEMPLATE).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -26,10 +19,12 @@ _Include any links here that might be helpful for people reviewing your PR. If t
...
 ## Output from acceptance tests
 
-_Please run the full suite of acceptance tests locally and include the output here._
+_Please run applicable acceptance tests locally and include the output here. If you are an external
+contributor, your contribution(s) will first be reviewed before running them against the project's
+CI pipeline._
 
 ```
-$ make testacc
+$ go test -v ./... -run TestAccTFEWorkspace # runs all workspace tests
```
Comment (radditude) [inline .github/pull_request_template.md:27]: Nitpick: I don't think this command will actually run the acceptance tests unless the user has `TF_ACC=1` set!

```suggestion
$ TESTARGS="-run TestAccTFEWorkspace" make testacc 
```

Comment (radditude) [inline .github/pull_request_template.md:27]: Nitpick: I don't think this command will actually run the acceptance tests unless the user has `TF_ACC=1` set!

```suggestion
$ TESTARGS="-run TestAccTFEWorkspace" make testacc 
```

Comment (radditude) [inline .github/pull_request_template.md:27]: Nitpick: I don't think this command will actually run the acceptance tests unless the user has `TF_ACC=1` set!

```suggestion
$ TESTARGS="-run TestAccTFEWorkspace" make testacc 
```

Comment (chrisarcand) [reply .github/pull_request_template.md:27]: Oh oops, yeah I just always have that set TBH; your version is must more practical!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -26,10 +19,12 @@ _Include any links here that might be helpful for people reviewing your PR. If t
 
 ## Output from acceptance tests
 
-_Please run the full suite of acceptance tests locally and include the output here._
+_Please run applicable acceptance tests locally and include the output here. If you are an external
+contributor, your contribution(s) will first be reviewed before running them against the project's
+CI pipeline._
```
Comment (radditude) [inline .github/pull_request_template.md:24]:
```suggestion
_Please run applicable acceptance tests locally and include the output here. Please see TESTS.md for help running the acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._
```

**Review by radditude — APPROVED** [review]
Looks good! I'm not sure why Github decided to make those all separate comments instead of one PR review, but w/e

**Pull request merged**